<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav)" />
  <arg name="name" default="" />
  <arg name="profiler" default="$(optenv PROFILER false)" />

  <arg name="swap_garmins" default="$(optenv SWAP_GARMINS false)" />

    <!-- will it run using GNU debugger? -->
  <arg name="DEBUG" default="false" />
  <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>

    <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_llcp_manager" />
  <arg name="n_threads" default="8" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <group ns="$(arg UAV_NAME)">

    <!-- launch the nodelet -->
    <node pkg="nodelet" type="nodelet" name="garmin_gimbal_llcp" args="$(arg nodelet) garmin_gimbal_llcp/GarminGimbalLlcp $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <param name="uav_name" type="string" value="$(arg UAV_NAME)"/>

      <param name="enable_profiler" type="bool" value="$(arg profiler)" />
      <param name="use_timeout" value="false"/>
      <param name="swap_garmins" value="$(arg swap_garmins)"/>
     
      <remap from="~llcp_in" to="llcp_2/received_message" />
      <remap from="~llcp_out" to="llcp_2/send_message" />

      <remap from="~range" to="/$(arg UAV_NAME)/garmin/range" />
      <remap from="~range_up" to="/$(arg UAV_NAME)/garmin/range_up" />
      <remap from="~gimbal_command" to="/$(arg UAV_NAME)/tarot_gimbal/gimbal_command" />

    </node>

  </group>

  <include file="$(find mrs_llcp_ros)/launch/llcp_2.launch" >

    <arg name="baudrate" value="115200" />
    <arg name="portname" value="/dev/arduino" />

  </include>

</launch>
